### PR TITLE
Change per etype fanout to canonical etype

### PR DESCRIPTION
**Pull request by classicsong**
*Issue #, if available:*
#256 

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: python/graphstorm/config/argument.py [hunk 1]
```diff
@@ -480,15 +480,19 @@ def _check_fanout(self, fanout, fot_name):
             else:
                 # Fanout in format of
                 # etype2:20@etype3:20@etype1:20,etype2:10@etype3:4@etype1:2
+                # Each etype should be a canonical etype in format of
+                # srcntype_relation_dstntype
 
-                fanout = [{k.split(":")[0]: int(k.split(":")[1]) \
+                fanout = [{tuple(k.split(":")[0].split('_')): int(k.split(":")[1]) \
```
Comment (zheng-da) [inline python/graphstorm/config/argument.py:486]: what is this underscore?

Comment (classicsong) [reply python/graphstorm/config/argument.py:486]: Any other suggestions? The `,` and `:` are used. how about `;` ?

Comment (zheng-da) [reply python/graphstorm/config/argument.py:486]: let's not use underscore. we can potentially use `|`, '/'. '_' is commonly used in names.

File: tests/end2end-tests/graphstorm-ec/mgpu_test.sh [hunk 2]
```diff
@@ -198,7 +198,7 @@ rm -fr /data/gsgnn_ec_lm/*
...
+python3 -m graphstorm.run.gs_edge_classification --workspace $GS_HOME/training_scripts/gsgnn_ep/ --num-trainers $NUM_TRAINERS --num-servers 1 --num-samplers 0 --part-config /data/movielen_100k_multi_label_ec/movie-lens-100k.json --ip-config ip_list.txt --ssh-port 2222 --cf ml_ec.yaml --exclude-training-targets True --multilabel true --num-classes 6 --node-feat-name feat --use-mini-batch-infer false --topk-model-to-save 1  --save-embed-path /data/gsgnn_ec/emb/ --save-model-path /data/gsgnn_ec/ --save-model-frequency 1000 --decoder-edge-feat user,rating,movie:rate --fanout 'user_rating_movie:4@movie_rating-rev_user:5,user_rating_movie:2@movie_rating-rev_user:2' --num-layers 2
```
Comment (zheng-da) [inline tests/end2end-tests/graphstorm-ec/mgpu_test.sh:201]: you are still using underscore?

**Review by zheng-da — APPROVED** [review]

File: tests/end2end-tests/graphstorm-ec/mgpu_test.sh (hunk 2 above)
Comment (classicsong) [reply tests/end2end-tests/graphstorm-ec/mgpu_test.sh]: Updated

**Pull request merged**